### PR TITLE
Add external loader plugin from webpack-contrib

### LINKED ISSUE #7: Add the external-loader-plugin
Add the external-loader-plugin from https://github.com/dojo/webpack-contrib. Some potential complications - we no longer use the copy-webpack-plugin, as the files copied are not versioned. So if we are to use it for the external-loader-plugin, we need to decide what the versioning strategy is.

### PR DESCRIPTION
**Pull request by mwistrand**
Resolves #7. Requires a patch release of `webpack-contrib`, which this PR makes up for by directly  adding dependencies to `package.json` and `test-app/package.json` (these will be removed prior to merging).

Adds the `ExternalLoaderPlugin` from `@dojo/webpack-contrib`, allowing external dependencies that cannot be loaded with webpack to be injected from outside the build. There is a nearly identical implementation in [`@dojo/cli-build-webpack`](https://github.com/dojo/cli-build-webpack/blob/master/src/webpack.config.ts#L118-L137), but with two notable differences:

- There is no default module type. Previously, the default type was "amd", but this currently does not work as expected with webpack 3 (see webpack/webpack#6302)
- Exact mids can be specified. For example, `dojo/_base/lang` can be excluded directly without needing to exclude all of `dojo` or `dojo/_base`.

### CODE REVIEW

**Comment by dylans** [comment]
Documentation such as https://github.com/dojo/cli-build-webpack#interop-with-external-libraries should probably be updated in this PR when this change is landed.

**Comment by agubler** [comment]
@maier49 @mwistrand There are two open PRs for supporting externals, this and #98 -  Do you think we could consolidate them down into a single PR and make any requested changes in the reviews?

**Comment by mwistrand** [comment]
Closing in favor of #98.

File: package.json [hunk 1]
```diff
@@ -112,6 +112,7 @@
     "express": "4.16.2",
     "extract-text-webpack-plugin": "3.0.2",
     "file-loader": "1.1.5",
+    "html-webpack-include-assets-plugin": "1.0.2",
```
Comment (matt-gadd) [inline package.json]: is this used?

**Pull request merged**